File: cpp/variables_and_types.cpp
// VARIABLES AND DATA TYPES
// Demonstrates basic variable declaration and data types in C++

#include <iostream>
using namespace std;

int main() {
    // Primitive data types
    int age = 25;
    char grade = 'A';
    float height = 5.8f;
    double salary = 50000.50;
    bool isActive = true;
    
    cout << "Age: " << age << endl;
    cout << "Grade: " << grade << endl;
    cout << "Height: " << height << endl;
    cout << "Salary: " << salary << endl;
    cout << "Is Active: " << isActive << endl;
    
    // Type casting
    int x = 10;
    double y = static_cast<double>(x);
    cout << "Casted value: " << y << endl;
    
    return 0;
}

File: java/variables_and_types.java
// VARIABLES AND DATA TYPES
// Demonstrates variable declaration and data types in Java

public class VariablesAndTypes {
    public static void main(String[] args) {
        // Primitive data types
        int age = 25;
        String name = "Alice";
        double height = 5.8;
        float salary = 50000.50f;
        boolean isActive = true;
        char grade = 'A';
        
        System.out.println("Age: " + age);
        System.out.println("Name: " + name);
        System.out.println("Height: " + height);
        System.out.println("Salary: " + salary);
        System.out.println("Is Active: " + isActive);
        System.out.println("Grade: " + grade);
        
        // Type casting
        int x = 10;
        double y = (double) x;
        System.out.println("Casted value: " + y);
        
        // String conversion
        int num = 42;
        String numStr = String.valueOf(num);
        System.out.println("Converted to String: " + numStr);
        
        // Using String.format
        String formatted = String.format("Value: %d, Name: %s", age, name);
        System.out.println(formatted);
    }
}
